### PR TITLE
fix:  create annotation run crashes if some but not all samples are labeled

### PR DESCRIPTION
**Pull request by oddeirikigland**
## What changes are proposed in this pull request?

Creating an annotation run works fine, but when you add new images to the dataset and want to create a new annotation run with all the images, both the new and old, labelstudio crashes. This PR fixes this since labels are only imported from samples that have labels. 

## How is this patch tested? If it is not, please explain why.

Versions:

fiftyone version: 0.17.2
Tested with both label-studio-sdk 0.0.14 and 0.0.15 

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [x] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## To reproduce:
```
d.annotate(
    anno_key=annotation_key,
    label_schema={
        gt_field: {
            "type": "keypoints",
            "classes": classes,
        }
    },
)
```
Where d is a fiftyone dataset and some of the samples in d have values in `gt_field` and some do not.

### CODE REVIEW

**Review by ehofesmann — CHANGES_REQUESTED** [review]
Thanks for catching this bug and making this PR!

This solution does work, but it may be better to still add the sample id to the `predictions` and `id_map`, just with empty labels.

This would allow `predictions` to still include the full result information about the image even if there are no labels to upload. Also, it can be useful to store information in the `id_map` about the exact samples that were uploaded even if they had no labels.

So basically instead of changing the lines here, I recommend updating the `export_label_to_label_studio()` and `_get_label_ids()` methods to handle `None` values. For eaxmple:
```python
def _get_label_ids(label):
    if label is None:
        return []
    elif isinstance(
...
```

File: fiftyone/utils/labelstudio.py [hunk 1]
```diff
@@ -636,7 +636,11 @@ def export_label_to_label_studio(label, full_result=None):
         a dictionary or a list in Label Studio format
     """
     # TODO model version and model score
-    if _check_type(label, fol.Classification, fol.Classifications):
+    if label is None:
+        result_value = {}
+        ls_type = ""
+        ids = [0]
```
Comment (oddeirikigland) [inline fiftyone/utils/labelstudio.py:642]: which values should ls_type and ids have?

**Comment by oddeirikigland** [comment]
Changed as requested @ehofesmann

File: fiftyone/utils/labelstudio.py (hunk 1 above)
Comment (ehofesmann) [reply fiftyone/utils/labelstudio.py:642]:
```suggestion
        ls_type = None
        ids = []
```

`ids` will need to be an empty list in this case. Since `ids` and `result_value` are empty, `ls_type` will not be used. I would just set it to `ls_type = None`.

**Review by ehofesmann — APPROVED** [review]
Amazing! Thanks for this fix @oddeirikigland !
